Marlin/Conditionals_post.h

```diff
@@ -379,6 +379,152 @@
+      #define USE_XMAX_PLUG
+    #elif X2_USE_ENDSTOP == _YMIN_
+      #define USE_YMIN_PLUG
+    #elif X2_USE_ENDSTOP == _YMAX_
+      #define USE_YMAX_PLUG
+    #elif X2_USE_ENDSTOP == _ZMIN_
+      #define USE_ZMIN_PLUG
+    #elif X2_USE_ENDSTOP == _ZMAX_
+      #define USE_ZMAX_PLUG
+    #endif
+    #if X_HOME_DIR > 0
+      #if X2_USE_ENDSTOP == _XMIN_
+        #define X2_MAX_ENDSTOP_INVERTING X_MIN_ENDSTOP_INVERTING
+        #define X2_MAX_PIN X_MIN_PIN
+      #elif X2_USE_ENDSTOP == _XMAX_
+        #define X2_MAX_ENDSTOP_INVERTING X_MAX_ENDSTOP_INVERTING
+        #define X2_MAX_PIN X_MAX_PIN
+      #elif X2_USE_ENDSTOP == _YMIN_
+        #define X2_MAX_ENDSTOP_INVERTING Y_MIN_ENDSTOP_INVERTING
+        #define X2_MAX_PIN Y_MIN_PIN
+      #elif X2_USE_ENDSTOP == _YMAX_
+        #define X2_MAX_ENDSTOP_INVERTING Y_MAX_ENDSTOP_INVERTING
+        #define X2_MAX_PIN Y_MAX_PIN
+      #elif X2_USE_ENDSTOP == _ZMIN_
+        #define X2_MAX_ENDSTOP_INVERTING Z_MIN_ENDSTOP_INVERTING
+        #define X2_MAX_PIN Z_MIN_PIN
+      #elif X2_USE_ENDSTOP == _ZMAX_
+        #define X2_MAX_ENDSTOP_INVERTING Z_MAX_ENDSTOP_INVERTING
+        #define X2_MAX_PIN Z_MAX_PIN
+      #else
+        #define X2_MAX_ENDSTOP_INVERTING false
+      #endif
+    #else
+      #if X2_USE_ENDSTOP == _XMIN_
+        #define X2_MIN_ENDSTOP_INVERTING X_MIN_ENDSTOP_INVERTING
+        #define X2_MIN_PIN X_MIN_PIN
+      #elif X2_USE_ENDSTOP == _XMAX_
+        #define X2_MIN_ENDSTOP_INVERTING X_MAX_ENDSTOP_INVERTING
+        #define X2_MIN_PIN X_MAX_PIN
+      #elif X2_USE_ENDSTOP == _YMIN_
+        #define X2_MIN_ENDSTOP_INVERTING Y_MIN_ENDSTOP_INVERTING
+        #define X2_MIN_PIN Y_MIN_PIN
+      #elif X2_USE_ENDSTOP == _YMAX_
+        #define X2_MIN_ENDSTOP_INVERTING Y_MAX_ENDSTOP_INVERTING
+        #define X2_MIN_PIN Y_MAX_PIN
+      #elif X2_USE_ENDSTOP == _ZMIN_
+        #define X2_MIN_ENDSTOP_INVERTING Z_MIN_ENDSTOP_INVERTING
+        #define X2_MIN_PIN Z_MIN_PIN
+      #elif X2_USE_ENDSTOP == _ZMAX_
+        #define X2_MIN_ENDSTOP_INVERTING Z_MAX_ENDSTOP_INVERTING
+        #define X2_MIN_PIN Z_MAX_PIN
+      #else
+        #define X2_MIN_ENDSTOP_INVERTING false
+      #endif
+    #endif
+  #endif
+
+  // Is an endstop plug used for the X2 endstop?
+  #define IS_X2_ENDSTOP(A,M) (ENABLED(X_DUAL_ENDSTOPS) && X2_USE_ENDSTOP == _##A##M##_)
+
+  /**
+   * Y_DUAL_ENDSTOPS endstop reassignment
+   */
+  #if ENABLED(Y_DUAL_ENDSTOPS)
+    #define _XMIN_ 100
+    #define _YMIN_ 200
+    #define _ZMIN_ 300
+    #define _XMAX_ 101
+    #define _YMAX_ 201
+    #define _ZMAX_ 301
+    #if Y2_USE_ENDSTOP == _XMIN_
+      #define USE_XMIN_PLUG
+    #elif Y2_USE_ENDSTOP == _XMAX_
+      #define USE_XMAX_PLUG
+    #elif Y2_USE_ENDSTOP == _YMIN_
+      #define USE_YMIN_PLUG
+    #elif Y2_USE_ENDSTOP == _YMAX_
+      #define USE_YMAX_PLUG
+    #elif Y2_USE_ENDSTOP == _ZMIN_
+      #define USE_ZMIN_PLUG
+    #elif Y2_USE_ENDSTOP == _ZMAX_
+      #define USE_ZMAX_PLUG
+    #endif
```

Comment:
It's friendly to set these on behalf of users, but unnecessary. We insist that users set these manually in the `Configuration.h` file to prove that they're paying attention.